### PR TITLE
Make the fluffy kitty button meow

### LINKED ISSUE #763: The cat should meow when you click on it.
### Role

I am but a lonely denizen of this cold, cold world.

### Suggestion

Meow.

### Benefit

Whenever I click the cat button and a cat appears, I feel a deep, deep longing in my soul that would only be filled if the cat button were only to meow. The lack of a meow is soul-crushing, so much so that I sometimes keel over in pain at the solemn silence that occurs when a cat appears but does not meow. Meow. That is all I need to hear, just a meow. Meow. Meow. Meow. Meow.

### This suggestion is unique

- [X] I have searched the issue tracker and did not find an issue describing my suggestion, especially not one that has been rejected.

### You may use the editor below to elaborate further.

The main reason that MultiMC avoided the meow of the cat (according to a conversation that I had 5 months ago with the MultiMC maintainer, poorly recalled from memory) was because Qt5.5's built-in sound playback crackled and clicked the meow. However, I believe that the Qt (5.12?) devs have heard my furry plight (Not a furry. Just referring to the lovely sound a furry creature called a cat makes when you click on it) and have allowed you wonderful developers to utilize the QSound class to make the beautiful noise I have been longing for for so, so, long. A sound that would make this cold place feel just a bit more like home.
Sincerely,
asolidtime

### PR DESCRIPTION
**Pull request by TheEssem**
Fixes #763, arguably the most requested feature of all time. This uses QSound to play back the audio which is loaded at application start.

Link to the original CC0-licensed sound: https://freesound.org/people/spacether/sounds/385892/

### CODE REVIEW

**Comment by DioEgizio** [comment]
To not make ci fail, add qtmultimedia5-dev to https://github.com/PolyMC/PolyMC/blob/309013efb3e148310ba4802ee946a55c01d1d56e/.github/workflows/build.yml#L124

**Comment by DioEgizio** [comment]
doesn't work for me, doesn't meow D:

**Comment by timoreo22** [comment]
to fix the ubuntu appimage build, adding `libqt5multimedia5-plugins` as another package to install would probably fix it
still on the same line
https://github.com/PolyMC/PolyMC/blob/309013efb3e148310ba4802ee946a55c01d1d56e/.github/workflows/build.yml#L124

**Comment by ryanccn** [comment]
jump scare

**Comment by jn64** [comment]
The original sound is a bit loud and has some harshness. I've normalised it so it is in line with the loudness of freedesktop bell alert sound (probably the most common alert sound on Linux desktops, just picked as an example as I do not have access to other OS) and filtered some of the high frequencies.

[meow-normalised-filtered3.zip](https://github.com/PolyMC/PolyMC/files/8877622/meow-normalised-filtered3.zip) - please use this if it is acceptable

I did not match the change you made to the length of the file as I don't know if that was intentional (the sound in the PR has a negligible ~11ms of silence added to the front of the original)

**Comment by forkiesassds** [comment]
Why? Why do we need to add a dependency just for one joke feature?

**Comment by Regular-Baf** [comment]
> Why? Why do we need to add a dependency just for one joke feature?

Because it's awesome. God bless you @TheEssem

**Review by Regular-Baf — APPROVED** [review]
No clue what any of this does but it has my vote.

**Review by zzcolby27 — APPROVED** [review]
Based and catpilled

**Review by ryanccn — APPROVED** [review]
It's a joke!

**Review by DioEgizio — CHANGES_REQUESTED** [review]
Doesn't work D:

**Comment by Un1q32** [comment]
merge wen

**Comment by TheEssem** [comment]
@timoreo22 @jn64 Thanks for the tips, fixed.

**Comment by Scrumplex** [comment]
hmm. not sure if we want to pull Qt Multimedia just for a meow

**Comment by TheEssem** [comment]
> hmm. not sure if we want to pull Qt Multimedia just for a meow

Was kinda concerned about that as well, how do you suggest it be done instead? The original issue mentioned it and so that's what I went for

**Comment by LoganDark** [comment]
Meow :)

**Comment by forkiesassds** [comment]
> hmm. not sure if we want to pull Qt Multimedia just for a meow

Exactly. Why do we need a multimedia library just for a single feature?

**Comment by asolidtime** [comment]
If we add a config option to meow (and show a notification) when a modpack or game version finishes downloading, would that better justify pulling qtmultimedia?  /hj
  
  
  
  
  
i didn't realize qsound needed an extra dependency sorry

**Comment by asolidtime** [comment]
oh, also, wouldn't it be a bit better if it only meowed when the cat was shown, not when it's hidden?

File: CMakeLists.txt [hunk 1]
```diff
@@ -151,7 +151,7 @@ add_custom_target(tcversion echo "\\#\\#teamcity[setParameter name=\\'env.LAUNCH
 # Find the required Qt parts
 if(Launcher_QT_VERSION_MAJOR EQUAL 5)
     set(QT_VERSION_MAJOR 5)
-    find_package(Qt5 REQUIRED COMPONENTS Core Widgets Concurrent Network Test Xml)
+    find_package(Qt5 REQUIRED COMPONENTS Core Widgets Concurrent Multimedia Network Test Xml)
```
Comment (kthchew) [inline CMakeLists.txt:154]: The [uninstaller section](https://github.com/PolyMC/PolyMC/blob/707a68cb4f71c8de5476064215f49ead011a9038/program_info/win_install.nsi.in#L170-L238) in the script to generate the Windows (un)installer should be updated to remove any file(s) associated with the new multimedia dependency (`Qt5Multimedia.dll`, I think).

```nsis
Delete $INSTDIR\Qt5Multimedia.dll
```

**Review by kthchew — CHANGES_REQUESTED** [review]

**Comment by txtsd** [comment]
> hmm. not sure if we want to pull Qt Multimedia just for a meow

How about we leave it in as an easter egg? So if anyone already has Qt Multimedia, it will play the sound. Otherwise the behavior is unchanged.
@Scrumplex

**Comment by ZekeZDev** [comment]
I like that idea

**Review by jamieslyman — CHANGES_REQUESTED** [review]
While it would be funny to keep it, I think that `ctaMeow` should be renamed into `catMeow` unless there's a reason not to.

**Comment by ZekeZDev** [comment]
I think we can all agree while nice this isnt useful and just adds unessecary dependencies. I am going to close this issue but remind me again for 1st of april next year might use it in a custom build.

**Pull request closed without merging**